addons/wolf_modules/tasksystem/fn_updateActiveTasks.sqf

```diff
@@ -13,28 +13,47 @@
  * Public: No
  */
 
-
-
 {
-	// _x: TaskInstance
-	// Check if complete
-	private _isCompleted = _x call ["completionCondition"];
-	if (!_isCompleted) then { continue; }; // Don't care about uncompleted tasks
-
-	diag_log ["task completed", (_x get "taskType") get "taskType"];
-
-	// Task is completed, remove it from active, finish it, check for continuation 
-	wolf_tasksystem_activeTasks deleteAt (wolf_tasksystem_activeTasks find _x);
-	_x call ["onComplete"];
-
-	// Continuation? 
-	private _continuation = _x call ["getContinuationType"];
-
-	if (isNil "_continuation") then { continue; }; // No continuation, we're done with this
-
-	(_continuation call wolf_tasksystem_fnc_selectTaskType) params ["_newTaskType", "_targetLocation"];
-
-	if (isNil "_newTaskType") then { continue; }; // Couldn't find anything to continue with
-
-	[_newTaskType, _targetLocation, _x] call wolf_tasksystem_fnc_startNewTask;
-} forEachReversed wolf_tasksystem_activeTasks; // Reversed because we're deleting elements
\ No newline at end of file
+    // _x: TaskInstance
+    private _taskID = _x get "taskID";
+    private _taskType = (_x get "taskType") get "taskType";
+
+    if (_x call ["completionCondition"]) then {
+        // Task completed
+        diag_log ["Task completed", _taskType];
+
+        // Remove task from active list and execute completion routine
+        wolf_tasksystem_activeTasks deleteAt (wolf_tasksystem_activeTasks find _x);
+        _x call ["onComplete"];
+
+        // Check for continuation task
+        private _continuation = _x call ["getContinuationType"];
+        if (!isNil "_continuation") then {
+            (_continuation call wolf_tasksystem_fnc_selectTaskType) params ["_newTaskType", "_targetLocation"];
+            if (!isNil "_newTaskType") then {
+                [_newTaskType, _targetLocation, _x] call wolf_tasksystem_fnc_startNewTask;
+            };
+        };
+
+    } else {
+        if (_x call ["failureCondition"]) then {
```

Comment:
you can use if (..) exitWith {} instead of nested.
but this looks okay enough/readable enough